rotkehlchen/chain/ethereum/modules/hop/decoder.py

```diff
@@ -1,70 +1,81 @@
+)
+from rotkehlchen.chain.evm.constants import DEFAULT_TOKEN_DECIMALS
+from rotkehlchen.chain.evm.decoding.hop.constants import CPT_HOP, HOP_CPT_DETAILS
+from rotkehlchen.chain.evm.decoding.hop.decoder import HopCommonDecoder
 from rotkehlchen.chain.evm.decoding.structures import (
     DEFAULT_DECODING_OUTPUT,
     DecoderContext,
     DecodingOutput,
 )
 from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
-from rotkehlchen.chain.evm.types import string_to_evm_address
 from rotkehlchen.constants.assets import A_ETH
-from rotkehlchen.fval import FVal
 from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
-from rotkehlchen.types import ChecksumEvmAddress
-from rotkehlchen.utils.misc import from_wei, hex_or_bytes_to_address, hex_or_bytes_to_int
+from rotkehlchen.types import ChainID, ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
 
-# https://github.com/hop-protocol/hop/blob/develop/packages/core/src/addresses/mainnet.ts
-ETH_BRIDGE = string_to_evm_address('0xb8901acB165ed027E32754E0FFe830802919727f')
-
-TRANSFER_TO_L2 = b'\n\x06\x07h\x8c\x86\xec\x17u\xab\xcd\xba\xb7\xb3::5\xa6\xc9\xcd\xe6w\xc9\xbe\x88\x01P\xc21\xcck\x0b'  # noqa: E501
+from .constants import BRIDGES
 
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.user_messages import MessagesAggregator
 
-# Probably this should go somewhere more central eventually
-# and add other networks too. Added only hop supported ones now
-chainid_to_name = {
-    1: 'Ethereum Mainnet',
-    10: 'Optimism',
-    100: 'Gnosis Chain',
-    137: 'Polygon POS',
-    42161: 'Arbitrum One',
-}
+TRANSFER_TO_L2 = b'\n\x06\x07h\x8c\x86\xec\x17u\xab\xcd\xba\xb7\xb3::5\xa6\xc9\xcd\xe6w\xc9\xbe\x88\x01P\xc21\xcck\x0b'  # noqa: E501
 
 
-class HopDecoder(DecoderInterface):
+class HopDecoder(HopCommonDecoder):
+    def __init__(
+            self,
+            ethereum_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=ethereum_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+            bridges=BRIDGES,
+        )
 
-    def _decode_send_eth(self, context: DecoderContext) -> DecodingOutput:
+    def _decode_send_to_l2(self, context: DecoderContext) -> DecodingOutput:
         if context.tx_log.topics[0] != TRANSFER_TO_L2:
             return DEFAULT_DECODING_OUTPUT
 
-        chain_id = hex_or_bytes_to_int(context.tx_log.topics[1])
+        if (bridge := self.bridges.get(context.tx_log.address)) is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        chain_id_raw = hex_or_bytes_to_int(context.tx_log.topics[1])
+        chain_id = ChainID.deserialize(chain_id_raw)
```

Comment:
```suggestion
        chain_id = ChainID.deserialize(hex_or_bytes_to_int(context.tx_log.topics[1]))
```

Reply:
What happens if we don't track the chain? I believe this logic shouldn't fail

Reply:
Adding a classmethod to return f'Unknown Chain ID: {value}' if not tracked

Reply:
Changing the label to empty, if error on chain id.